### PR TITLE
Allow admins to moderate anyone, including themselves or other admins

### PR DESCRIPTION
**Pull request by Scott-Caldwell**
I noticed that the bot was preventing admins from deleting infractions from themselves, which seemed odd. It would allow infractions to get into a "stuck" state where nobody could rescind or delete them. Besides that, the point of the admin permission is that you can do whatever you want short of deleting the server, so this PR rectifies that by ignoring admins in the rank validation.

Another potential option might be to create a claim for avoiding the rank check and giving that claim to the admin role.

I could go either way on this one.

### CODE REVIEW

**Comment by Scott-Caldwell** [comment]
Like so?

File: Modix.Services/Core/AuthorizationService.cs [hunk 1]
```diff
@@ -28,6 +28,11 @@ public interface IAuthorizationService
         /// </summary>
         ulong? CurrentGuildId { get; }
 
+        /// <summary>
+        /// Indicates whether the current user (if any) should be ignored during authorization checks.
+        /// </summary>
+        bool? CurrentUserIgnoresAuthorization { get; }
```
Comment (jmazouri) [inline Modix.Services/Core/AuthorizationService.cs]: I would just make this a regular `bool`. It'll always be false when there's no authenticated user, which shouldn't break any other logic - if there's no user, they can't ignore authorization.

File: Modix.Services/Moderation/ModerationService.cs [hunk 2]
```diff
@@ -628,8 +628,11 @@ private async Task<IEnumerable<GuildRoleBrief>> GetRankRolesAsync()
                 }))
                 .Select(r => r.Role);
 
-        private async Task RequireSubjectRankLowerThanModeratorRank(ulong guildId, ulong subjectId)
+        private async Task RequireSubjectRankLowerThanModeratorRankAsync(ulong guildId, ulong subjectId)
         {
+            if (AuthorizationService.CurrentUserIgnoresAuthorization.Value)
```
Comment (jmazouri) [inline Modix.Services/Moderation/ModerationService.cs]: See above - probably doesn't need to be nullable. Also, you should call `AuthorizationService.RequireAuthenticatedUser();` right before (apologies if it already gets called somewhere else in the stack).

**Comment by JakenVeina** [comment]
I mean, yes, essentially, but I was not really intending for it to be included in this PR. The concept of a catch-all loophole to the auth system is rather risky, and I'd like to give all the maintainers time to consider it, and try and think up any ways it could be abused. Leave it in, if you want, but don't expect this to get merged any time soon.

**Comment by Scott-Caldwell** [comment]
Ah, I see. Sorry for the confusion. I'll just take it out then. Probably not worth keeping it in right now

**Pull request merged**